pkg/controller/seed-controller-manager/initialmachinedeployment/controller.go

```diff
@@ -85,7 +86,12 @@ func Add(ctx context.Context, mgr manager.Manager, numWorkers int, workerName st
 		return fmt.Errorf("failed to create controller: %w", err)
 	}
 
-	if err := c.Watch(&source.Kind{Type: &kubermaticv1.Cluster{}}, &handler.EnqueueRequestForObject{}); err != nil {
+	predicate := predicateutil.MultiFactory(predicateutil.TrueFilter, func(o ctrlruntimeclient.Object) bool {
+		_, exists := o.GetAnnotations()[v1.InitialMachineDeploymentRequestAnnotation]
+		return exists
+	}, predicateutil.TrueFilter)
+
+	if err := c.Watch(&source.Kind{Type: &kubermaticv1.Cluster{}}, &handler.EnqueueRequestForObject{}, predicate); err != nil {
```

Comment:
even though this was discovered during working on the Applications PR, this changes logic for the machinedeployment-controller. It should be separated from the Application Logic. Otherwise we are going to have a hard time code debugging in case this breaks something

Reply:
We have unit tests to debug this and E2E test as well. I don't see it as an issue since it was a very minor refactor.